### DISCUSSION TITLE
Bash prompt isn't being coloured

### DISCUSSION THREAD

**Discussion by EnronEvolved** [ghostty-org/ghostty#3537] [Q&A] [answered]
All other terminal colours seem to be working, but my bash prompt is not displaying in colour, despite doing so in Alacritty and Gnome Terminal.

I'm running Ghostty on Linux Mint 21.3, built with `zig build -Dgtk-adwaita=false -Doptimize=ReleaseFast -p $HOME/.local`.

This is what Ghostty puts on stdout:

```
info: ghostty version=1.0.1-main+a8e5eef1
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=8.4.0
info: dependency fontconfig=21402
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=en_GB.UTF-8
info(gtk): GTK version build=4.6.9 runtime=4.6.9
info: reading configuration file path=/home/…/.config/ghostty/config
info(config): default shell source=env value=/bin/bash
info(sentry): sentry envelope does not contain crash, discarding
"vulkan-disable" is only available when building GTK with G_ENABLE_DEBUG. See GDK_DEBUG=help
error(gtk): unable to get current color scheme: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such method “ReadOne”
info(grid): loaded OpenGL 4.6
info(font_shared_grid_set): font regular: DejaVu Sans Mono
info(font_shared_grid_set): font bold: DejaVu Sans Mono Bold
info(font_shared_grid_set): font italic: DejaVu Sans Mono Oblique
info(font_shared_grid_set): font bold_italic: DejaVu Sans Mono Bold Oblique
info(io_exec): found Ghostty resources dir: /home/…/.local/share/ghostty
info(io_exec): shell integration automatically injected shell=termio.shell_integration.Shell.bash
warning(gtk): unimplemented action=apprt.action.Action.Key.cell_size
warning(gtk): unimplemented action=apprt.action.Action.Key.size_limit
info(io_exec): started subcommand path=/bin/sh pid=130781
info(io_exec): subcommand cgroup=-
error(gtk): unable to get current color scheme: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such method “ReadOne”
info(grid): reallocating GPU buffer old=0 new=38
```

**Comment by jmckillican** [comment]
this is likely related to terminfo, a quick fix is setting term = xterm-256color in your ghostty config, more details here:

https://ghostty.org/docs/help/terminfo

**Reply by EnronEvolved** [reply]
Can confirm that setting `term = xterm-256color` (_not_ `color256`) makes bash behave itself, though `infocmp xterm-ghostty` does produce a terminfo entry (at `~/.local/share/terminfo/x/xterm-ghostty`). I'll ask the bash people about it.

**Comment by christiangnrd** [comment, marked as answer]
I spent hours yesterday trying to figure out why this was still happening after copying over `xterm-ghostty` to my remote servers running Ubuntu but not other distros. Turns out my `.bashrc` had the following line:
```
# set a fancy prompt (non-color, unless we know we "want" color)
case "$TERM" in
    xterm-color|*-256color) color_prompt=yes;;
esac

# uncomment for a colored prompt, if the terminal has the capability; turned
# off by default to not distract the user: the focus in a terminal window
# should be on the output of commands, not on the prompt
#force_color_prompt=yes

```

I changed it to 
```
# set a fancy prompt (non-color, unless we know we "want" color)
case "$TERM" in
    xterm-color|*-256color|*ghostty) color_prompt=yes;;
esac
```
but you could probably uncomment (or add) `force_color_prompt=yes` if this is indeed the cause of your issue.

**Reply by EnronEvolved** [reply]
This is it. It seems to be in the default Ubuntu `.bashrc`, so I expect damn near everyone trying to run Ghostty on Ubuntu (or any of its derivatives) will have this problem. Notably, Alacritty lies and identifies itself as `xterm-256color`.

[It seems to be a perennial problem](https://bugs.launchpad.net/ubuntu/+bugs?field.searchtext=bashrc+color+terminal&search=Search&field.status%3Alist=NEW&field.status%3Alist=INCOMPLETE_WITH_RESPONSE&field.status%3Alist=INCOMPLETE_WITHOUT_RESPONSE&field.status%3Alist=CONFIRMED&field.status%3Alist=TRIAGED&field.status%3Alist=INPROGRESS&field.status%3Alist=FIXCOMMITTED&field.assignee=&field.bug_reporter=&field.omit_dupes=on&field.has_patch=&field.has_no_package=&orderby=importance&start=0)…

**Reply by stevenlafl** [reply]
God, thank you.

**Comment by gtirloni** [comment]
In Fedora, `/etc/profile.d/bash-color-prompt.sh` has a similar check:

```
    # only activate for color terminals and if PS1 unchanged from bash or fedora defaults
    if [ '(' "$PS1" = "[\u@\h \W]\\$ " -o "$PS1" = "\\s-\\v\\\$ " ')' \
             -a '(' "${TERM: -5}" = "color" -o "${TERM}" = "linux" ')' \
             -o -n "${bash_prompt_color_force}" ]; then
```

**Comment by gaffpearce** [comment]
Thanks chrisiangnrd! I changed my .bashrc with the line: 

xterm-color|*-256color|*ghostty) color_prompt=yes;;

and it worked perfectly! Ghostty prompt is now in color.

**Reply by spullara** [reply]
Why would every Ubuntu user change their .bashrc because Ghostty makes a bad choice here?
